pkg/pool-manager/pod_pool.go

```diff
@@ -121,43 +120,42 @@ func (p *PoolManager) ReleasePodMac(podName string) error {
 	}
 
 	for _, macAddr := range macList {
-		delete(p.macPoolMap, macAddr)
-		log.Info("released mac from pod", "mac", macAddr, "pod", podName)
+		if isInRange, _ := p.IsMacInRange(macAddr); isInRange {
+			delete(p.macPoolMap, macAddr)
+			log.Info("released mac from pod", "mac", macAddr, "pod", podName)
+		}
 	}
 
-	delete(p.podToMacPoolMap, podName)
-	log.V(1).Info("removed pod from podToMacPoolMap", "pod", podName)
+	if _, exist := p.podToMacPoolMap[podName]; exist {
```

Comment:
why do we need the "if exist..."?
we check it here: 
https://github.com/k8snetworkplumbingwg/kubemacpool/pull/113/files#diff-f178743f06898ca1ae8f344f45065660R109

if we do need another check. why not use the `ok` var?
`if ok := p.podToMacPoolMap[podName]{ ...`

Reply:
Either the mac arrived from the pool, or set manually (and then it could be out of range and thus not inserted to podToMacPoolMap). Because of the latter - I check if it exists to avoid error..
correct me if I'm wrong, but since this map doesn't contain the mac, I can't ask if isMacInRange, so I have to check every time.

Regarding the syntax - I'm using the same syntax as [this](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/68e7721321f321911aa05148c68493dc44e6e555/pkg/pool-manager/pod_pool.go#L110). Did you mean change from "exist" to "ok"?

Reply:
>Either the mac arrived from the pool, or set manually (and then it could be out of range and thus not inserted to podToMacPoolMap). Because of the latter - I check if it exists to avoid error..

sorry but I am not sure I understand.
don't we check the same condition?  `if some_bool := p.podToMacPoolMap[podName]`
if the pod was there the first time. what could remove it?

Reply:
> if the pod was there the first time. what could remove it?

Please notice that we don't always insert the mac to macPoolMap & podToMacPoolMap in ([allocatePodRequestedMac()](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/f2e6bb8604dc6207ee538ca900d72505b9125f0a/pkg/pool-manager/pod_pool.go#L139))

since it is not inserted in case the mac was manually chosen & out of the pod's range - we cannot assume it's their now.

Reply:
I see. since I'm already asking if in the start of the [ReleasePodMac()](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/f2e6bb8604dc6207ee538ca900d72505b9125f0a/pkg/pool-manager/pod_pool.go#L109) 
Your'e totally right. will fix. (if not what you meant then please elaborate)

Reply:
exactly what I meant:)
Thank you.